python/lsst/ip/isr/isrTask.py

```diff
@@ -1770,6 +1771,115 @@ def run(self, ccdExposure, *, camera=None, bias=None, linearizer=None,
             outputStatistics=outputStatistics,
         )
 
+    def defineEffectivePtc(self, ptcDataset, detector, bfGains, overScans, metadata):
+        """Define an effective Photon Transfer Curve dataset
+        with nominal gains and noise.
+
+        Parameters
+        ------
+        ptcDataset : `lsst.ip.isr.PhotonTransferCurveDataset`
+            Inout Photon Transfer Curve dataset.
+        detector : `lsst.afw.cameraGeom.Detector`
+            Detector object.
+        bfGains : `dict`
+            Gains from running the brighter-fatter code.
+            A dict keyed by amplifier name for the detector
+            in question.
+        ovserScans : `list` [`lsst.pipe.base.Struct`]
+            List of overscanResults structures
+        metadata : `lsst.daf.base.PropertyList`
+            Exposure metadata to update gain and noise provenance.
+
+        Returns
+        -------
+        effectivePtc : `lsst.ip.isr.PhotonTransferCurveDataset`
+            PTC dataset containing gains and readout noise
+            values to be used throughout
+            Instrument Signature Removal.
+        """
+        amps = detector.getAmplifiers()
+        ampNames = [amp.getName() for amp in amps]
+        detName = detector.getName()
+        effectivePtc = PhotonTransferCurveDataset(ampNames, 'EFFECTIVE_PTC', 1)
+        boolGainMismatch = False
+
+        for amp, overscanResults in zip(amps, overScans):
+            ampName = amp.getName()
+            # Gain:
+            # Try first with the PTC gains.
+            gainProvenanceString = "amp"
+            if self.config.usePtcGains:
+                if ptcDataset is None:
```

Comment:
Can you move this check to ~L1348, with the other checks for `if doX and X is None`?  If the config doesn't match what was supplied, it's better to raise early.